### PR TITLE
✨ Support configuring proxy for addons

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/addon/templateagent/decorator.go [hunk 1]
```diff
@@ -329,6 +331,55 @@ func (d *imageDecorator) decorate(pod *corev1.PodTemplateSpec) error {
...
+			name:  "HTTPS_PROXY",
+			value: pc.HTTPSProxy,
+		})
+	}
+	if len(pc.NoProxy) > 0 {
+		keyValues = append(keyValues, keyValuePair{
+			name:  "NO_PROXY",
+			value: pc.NoProxy,
+		})
+	}
```
Comment (elgnay) [inline pkg/addon/templateagent/decorator.go:373]: I'm not entirely sure whether the environment variables for proxy settings should be uppercase, lowercase, or if both are supported.

Comment (zhujian7) [reply pkg/addon/templateagent/decorator.go:373]: It seems there is no standard and both uppercase and lowercase versions are used depending on the application

Comment (zhujian7) [reply pkg/addon/templateagent/decorator.go]: both uppercase and lowercase are added.

**Pull request merged**